pages/update-password.tsx

```diff
@@ -0,0 +1,125 @@
+import { inferSSRProps } from "@/lib/inferSSRProps";
+import { NextPageWithLayout } from "types";
+import { useFormik } from "formik";
+import axios from "axios";
+import * as Yup from "yup";
+import toast from "react-hot-toast";
+import { Card, InputWithLabel } from "@/components/ui";
+import { Button } from "react-daisyui";
+import { GetServerSidePropsContext } from "next";
+import { getSession } from "@/lib/session";
+import { getUserBySession } from "models/user";
+import { verifyPassword } from "@/lib/auth";
+import { useRouter } from "next/router";
+
+const UpdatePassword: NextPageWithLayout<inferSSRProps<typeof getServerSideProps>> = ({
+    user,
+}) => {
+    const router = useRouter();
+
+    const formik = useFormik({
+        initialValues: {
+            currentPassword: "",
+            newPassword: "",
+            confirmationPassword: ""
+        },
+        validationSchema: Yup.object().shape({
+            currentPassword: Yup.string().required(),
+            newPassword: Yup.string().required('Required'),
+            confirmationPassword: Yup.string().oneOf([Yup.ref('newPassword'), null], "Passwords don't match!")
+            .required('Required')
+        }),
+        onSubmit: async (values) => {
+            const { confirmationPassword } = values;
+            const currentPasswordIsValid = await verifyPassword(formik.values.currentPassword, String(user?.password));
```

Comment:
Should we do `verifyPassword` on the server side? Send all the 3 form values to the `/api/users` and do all the logic on the API route and display error message on the UI.